tutorials/inner_products/2_physical_properties.py

```diff
@@ -0,0 +1,357 @@
+print('\n FACE INNER PRODUCT MATRIX')
+print('- Number of faces              :', mesh.nF)
+print('- Dimensions of operator       :', str(mesh.nF), 'x', str(mesh.nF))
+print('- Number non-zero (isotropic)  :', str(Mf1.nnz))
+print('- Number non-zero (linear)     :', str(Mf2.nnz))
+print('- Number non-zero (anisotropic):', str(Mf3.nnz), '\n')
+
+print('\n EDGE INNER PRODUCT MATRIX')
+print('- Number of faces              :', mesh.nE)
+print('- Dimensions of operator       :', str(mesh.nE), 'x', str(mesh.nE))
+print('- Number non-zero (isotropic)  :', str(Me1.nnz))
+print('- Number non-zero (linear)     :', str(Me2.nnz))
+print('- Number non-zero (anisotropic):', str(Me3.nnz), '\n')
+
+
+#############################################################
+# Inverse
+# -------
+#
+# The final discretized system using the finite volume method may contain
+# the inverse of the inner-product matrix. Here we show how to call this
+# using the *invMat* keyword argument.
+#
+# For the isotropic and linear cases, the inner product matrix is diagonal.
+# As a result, its inverse can be easily formed. For the anisotropic case
+# however, we cannot expicitly form the inverse because the inner product
+# matrix contains a significant number of off-diagonal elements.
+#
+# For the isotropic case, we form M_inv and apply it to a vector using
+# the :math:`*` operator. For the anisotropic case, we must form the
+# inner product matrix and do a numerical solve.
+#
+
+# Create a small 3D mesh
+h = np.ones(5)
+mesh = TensorMesh([h, h, h])
+
+# Isotropic case: (nC, ) numpy array
+sig = np.random.rand(mesh.nC)
+Me1_inv = mesh.getEdgeInnerProduct(sig, invMat=True)
+Mf1_inv = mesh.getFaceInnerProduct(sig, invMat=True)
+
+# Linear case: (nC, dim) numpy array
+sig = np.random.rand(mesh.nC, mesh.dim)
+Me2_inv = mesh.getEdgeInnerProduct(sig, invMat=True)
+Mf2_inv = mesh.getFaceInnerProduct(sig, invMat=True)
+
+# Anisotropic case: (nC, 3) for 2D and (nC, 6) for 3D
+sig = np.random.rand(mesh.nC, 6)
+Me3 = mesh.getEdgeInnerProduct(sig)
+Mf3 = mesh.getFaceInnerProduct(sig)
+
+
+###########################################################################
+# Reciprocal Properties
+# ---------------------
+#
+# At times, the constitutive relation may be defined by the reciprocal of
+# a parameter (:math:`\rho`). Here we demonstrate how inner product matricies
+# can be formed using the keyword argument *invProp*. We will do this for a
+# single cell and plot the matrix elements. We can easily extend this to
+# a mesh comprised of many cells.
+#
+# In this case, the constitutive relation is given by:
+#
+# .. math::
+#     \vec{J} = \frac{1}{\rho} \vec{E}
+#
+# The inner product between a vector :math:`\\vec{v}` and the right-hand side
+# of the expression is given by:
+#
+# .. math::
+#     (\vec{v}, \rho^{-1} \vec{E} ) = \int_\Omega \vec{v} \cdot \rho^{-1} \vec{E} \, dv
+#
+# where the inner product is approximated using an inner product matrix
+# :math:`\mathbf{M_\rho}` as follows:
```

Comment:
This is a bit subtle notation-wise, in this case we want `:math:\mathbf{M_{\rho^{-1}}}` as we are forming the inner-product matrix for rho-inverse

Reply:
I thought the index of an index might be too small but I'll build the website and see if it look okay.